### PR TITLE
Switch to using the new `record_ecosystem_versions` endpoint.

### PR DESCRIPTION
**Pull request by jeffwidman**
This flips the `updater` to calling the new `record_ecosystem_versions`
endpoint.

Additionally, this flips the per-ecosystem implementations of
`package_manager_version` to this new `ecosystem_versions` hash.

Note that we expect the shape of the data to be slightly different in
the long term for these metrics... ie, we plan to record a
`min`/`max`/`raw` or something along those lines for the range of
versions supported by the manifest.

However, changing that will be done in separate PR's for easier
review/discussion/audit of those individual ecosystems.

Related:
* https://github.com/dependabot/dependabot-core/pull/7492
* https://github.com/dependabot/dependabot-core/pull/7516

### CODE REVIEW

**Comment by jeffwidman** [comment]
This is stacked on top of https://github.com/dependabot/dependabot-core/pull/7516 and only the final commit is new / needs review.

File: updater/lib/dependabot/api_client.rb
Comment (brrygrdn) [inline updater/lib/dependabot/api_client.rb:122]: Just making sure I've thought about this properly: this endpoint is new, so if someone pulls down an image with this change to GHES the service might not have that endpoint but that's fine as we've de-fanged the call unless an experiment is enabled in the [previous PR](https://github.com/dependabot/dependabot-core/pull/7516) and it will never been enabled for any version of GHES.

File: bin/dry-run.rb
Comment (brrygrdn) [inline bin/dry-run.rb:803]:
```suggestion
ecosystem_versions = fetcher.ecosystem_versions
puts "🎈 Ecosystem Versions log: #{ecosystem_versions}" unless ecosystem_versions.nil?
```

Style nit: Variable contractions always feel a little un-Ruby to me

File: updater/lib/dependabot/file_fetcher_command.rb
Comment (brrygrdn) [inline updater/lib/dependabot/file_fetcher_command.rb:27]:
```suggestion
ecosystem_versions = file_fetcher.ecosystem_versions
api_client.record_ecosystem_versions(ecosystem_versions) unless ecosystem_versions.nil?
```

Sorry, it's really a really minor but deeply engrained nitpick 😅

**Review by brrygrdn — APPROVED** [review]
This makes sense to me, 👍🏻

File: updater/lib/dependabot/api_client.rb
Comment (jeffwidman) [reply updater/lib/dependabot/api_client.rb:122]: Exactly.

File: updater/lib/dependabot/file_fetcher_command.rb
Comment (jeffwidman) [reply updater/lib/dependabot/file_fetcher_command.rb:27]: hah, I had that originally then decided it was too long. But I didn't want `versions` like it used to be as that felt too ambiguous. 👍 for this from me.

File: common/lib/dependabot/file_fetchers/base.rb
Comment (jeffwidman) [inline common/lib/dependabot/file_fetchers/base.rb:105]: This needs a code comment explaining the expected shape of the data, or possibly a doc somewhere for devs, but we're still noodling internally on what we want this to look like, so punting on those docs until later. 

Right now just want to get it swapped over to this new endpoint so can start prototyping actual code to see how my design ideas hold up to reality--ie, do we want to record more/different info then I expect.

File: updater/spec/dependabot/file_fetcher_command_spec.rb
Comment (jeffwidman) [inline updater/spec/dependabot/file_fetcher_command_spec.rb:112]: TODO before merge: try this to see if i can safely delete it... may need to first get some working ecosystems ported over to this though... like bundler, npm_and_yarn, etc.

File: updater/spec/dependabot/integration_spec.rb
Comment (jeffwidman) [inline updater/spec/dependabot/integration_spec.rb:62]: See https://github.com/dependabot/dependabot-core/pull/7517/files#r1252257873

File: updater/spec/dependabot/integration_spec.rb
Comment (jeffwidman) [inline updater/spec/dependabot/integration_spec.rb:83]: See https://github.com/dependabot/dependabot-core/pull/7517/files#r1252257873

**Comment by jeffwidman** [comment]
These test failures indicate may need to port the callers of this as part of this PR as well... or bury the package manager calls in private methods since the `ecosystem_versions()` will likely call both `package_manager_versions` and `programming_language_versions` :
```ruby
rspec './spec/dependabot/bundler/file_fetcher_spec.rb[1:1:1:5]' # Dependabot::Bundler::FileFetcher behaves like a dependency file fetcher the class doesn't define any additional public instance methods
```

**Pull request merged**
